third_party/terraform/resources/resource_compute_instance.go.erb

```diff
@@ -1389,51 +1423,61 @@ func resourceComputeInstanceUpdate(d *schema.ResourceData, meta interface{}) err
 					return opErr
 				}
 			}
-		}
-
-		if d.HasChange(prefix + ".alias_ip_range") {
-			rereadFingerprint := false
 
-			// Alias IP ranges cannot be updated; they must be removed and then added.
-			if len(instNetworkInterface.AliasIpRanges) > 0 {
-				ni := &computeBeta.NetworkInterface{
-					Fingerprint:     instNetworkInterface.Fingerprint,
-					ForceSendFields: []string{"AliasIpRanges"},
-				}
-				op, err := config.clientComputeBeta.Instances.UpdateNetworkInterface(project, zone, instance.Name, networkName, ni).Do()
-				if err != nil {
-					return errwrap.Wrapf("Error removing alias_ip_range: {{err}}", err)
-				}
-				opErr := computeOperationWaitTime(config, op, project, "updating alias ip ranges", d.Timeout(schema.TimeoutUpdate))
-				if opErr != nil {
-					return opErr
-				}
-				rereadFingerprint = true
+			//re-read fingerprint
+			instance, err = config.clientComputeBeta.Instances.Get(project, zone, instance.Name).Do()
+			if err != nil {
+				return err
 			}
+			instNetworkInterface = instance.NetworkInterfaces[i]
```

Comment:
why do we need to set instNetworkInterface again here?

Reply:
We've altered the instance. We need to update the reference so we have the correct fingerprint when we call UpdateNetworkInterface
see line 1467
```
networkInterface.Fingerprint = instNetworkInterface.Fingerprint
```